### ISSUE TITLE
[BUG] HTTP-GET module with option A=MD5 use Basic Auth instead of Digest Auth

### ISSUE THREAD

**Issue by bmos1** [vanhauser-thc/thc-hydra#888]
**Describe the bug**
Using hydra http-get module with the option A=MD5 (digest) should use Digest Auth only. However if you run hydra, you can find multiple Basic Auth attempts instead of Digest Auth only. **Important note**: It seems to be a threading issue, because the more tasks are running the more Basic Auth attempt can be seen. We have noticed a single Basic Auth attempt when running with options t 1 (single task). This is still not correct.

**To Reproduce**
  * Run hydra http-get module with option :A=MD5 
  * Use passwords.txt file with 100K entries

1. hydra -d -V -l root -P passwords.txt -F "http-get://10.10.10.10/cgi/login.cgi:A=MD5:F=You don't have permission to access this resource." &> hydra.debug
2. grep -n "Basic " hydra.debug
3. Authorization: **Basic** cm9vdDpiPDdwcnJSUDUj

**Expected behavior**
We expect to see Digest Auth only if we configure hydra http-get module with option :A=MD5. 
grep -n "Basic " hydra.debug | wc -l
0
grep -n "Digest " hydra.debug 
Authorization: **Digest** username="root"

**Screenshots**
None

**Desktop (please complete the following information):**
 - OS: Kali Linux 2023.3
 - hydra version v9.5 on Apple Silicon M1 / PRO

**Additional context**
None

**Comment by bmos1** [comment]
The following line in hydra-http.c leads to to the described behavior. The http_buf may be NULL due to currency issues.

```
 // we must reset this if buf is NULL and we do MD5 digest
  if (http_buf == NULL && http_auth_mechanism == AUTH_DIGESTMD5)
    http_auth_mechanism = AUTH_BASIC;
```
    
Here the unexpected result from option A=MD5 (Digest Auth) after a single run with 100K login attempts:

```
AUTH="Basic"; COUNT=$(grep ": $AUTH " hydra.debug | wc -l); echo "Found $COUNT $AUTH Auth"
Found 111054 Basic Auth

AUTH="Digest"; COUNT=$(grep ": $AUTH " hydra.debug | wc -l); echo "Found $COUNT $AUTH Auth" 
Found 12360 Digest Auth

```

**Comment by protechq88** [comment]
✌️ ✌️ ✌️

**Comment by bmos1** [comment]
You are right. If the server side responds with 403 Forbidden. The next request must start with new credentials and with Basic Auth. Anyways, if this request results in a 401 Unauthorized, hydra always should try the **same credentials** using configured MD5 Digest Auth.

If this is the case (untested), we can close the issue.
